app/src/main/java/com/example/habitsmasher/DatePickerDialogFragment.java

```diff
@@ -0,0 +1,72 @@
+package com.example.habitsmasher;
+
+
+import android.app.DatePickerDialog;
+import android.app.Dialog;
+import android.os.Bundle;
+
+import androidx.annotation.NonNull;
+import androidx.fragment.app.DialogFragment;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+
+/**
+ * Extraction of DatePickerDialog came from the following link:
+ *
+ * https://stackoverflow.com/a/34589270
+ * Post date: January 4, 2016
+ */
+
+public class DatePickerDialogFragment extends DialogFragment {
```

Comment:
I know it's technically a "fragment", but since this is mainly just a **pop-up**, I think renaming this class to just `DatePickerDialog` would flow better